tools/ci_status_report.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+import os
+import enum
+import argparse
+import collections
+import datetime
+import dataclasses
+import itertools
+
+import requests
+import matplotlib.pyplot as plt
+
+
+PROW_JOBS_STATUSES = "https://prow.ci.openshift.org/prowjobs.js?omit=annotations,labels,decoration_config,pod_spec"
+SLACK_FILES_UPLOAD = "https://slack.com/api/files.upload"
+TRACKED_REPOSITORIES = list(itertools.chain(
+    itertools.product(
+        ("openshift",), (
+            "assisted-test-infra",
+            "assisted-service",
+            "assisted-image-service",
+            "assisted-installer",
+            "assisted-installer-agent",
+        )),
+    itertools.product(
+        ("openshift-assisted",), (
+            "assisted-installer-deployment",
+            "assisted-events-scrape",
+        )),
+))
+
+
+class State(enum.Enum):
+    TRIGGERED = "triggered"  # job created but not yet scheduled
+    PENDING = "pending"  # job is currently running
+    SUCCESS = "success"  # job completed without any error
+    FAILURE = "failure"  # job completed with errors
+    ABORTED = "aborted"  # prow killed the job early (new commit pushed, perhaps)
+    ERROR = "error"  # job could not schedule (bad config, perhaps)
+
+
+class JobType(enum.Enum):
+    PRESUBMIT = "presubmit"
+    POSTSUBMIT = "postsubmit"
+    PERIODIC = "periodic"
+    BATCH = "batch"
+
+
+STATE_TO_COLOR = {
+    State.PENDING: "#FFD166",
+    State.ABORTED: "#118AB2",
+    State.FAILURE: "#EF476F",
+    State.ERROR: "pink",
+    State.SUCCESS: "#06D6A0",
+}
+
+
+@dataclasses.dataclass
+class Job:
+    type: JobType
+    name: str
+    organization: str
+    repository: str
+    base_ref: str
+    state: State
+    url: str
+    start_time: datetime.datetime
+    completion_time: datetime.datetime
+
+
+def filter_jobs(response):
```

Comment:
`items = response.json().get("items", {})`

`spec = document.get("spec", {})` 

To be more robust in the face of changed format

Also in the `job = Job` assignment, used `get` to avoid crashing on missing info

Reply:
(unless you want to crash on missing info which is also reasonable)

Reply:
I'll do that for the out scope, as it's indeed might be missing from the schema, and have everything around ``Job`` assignment with regular access, so we can crash on missing info